src/scenic/simulators/carla/simulator.py

```diff
@@ -169,3 +187,12 @@ def getProperties(self, obj, properties):
 			angularSpeed=utils.carlaToScenicAngularSpeed(currAngVel),
 		)
 		return values
+
+	def destroy(self):
+		for obj in self.objects:
+			if obj.carlaActor is not None:
+				obj.carlaActor.destroy()
+		if hasattr(self, "cameraManager"):
```

Comment:
`self.cameraManager` could be `None` if we get a spawn error while creating the ego object. I'd write `if self.render and self.cameraManager:` instead.

Reply:
Done